app/src/main/java/com/codesoom/assignment/services/TaskService.java

```diff
@@ -0,0 +1,119 @@
+package com.codesoom.assignment.services;
+
+import com.codesoom.assignment.common.exceptions.TaskNotFoundException;
+import com.codesoom.assignment.models.Task;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * 할 일(Task) 데이터를 가공하여 반환합니다.
+ */
+@Service
+public class TaskService {
+
+    private Long taskIdSeq = 0L;
+    private LinkedHashMap<Long, Task> tasksMap = new LinkedHashMap<>();
+
+    private Logger log = LoggerFactory.getLogger(TaskService.class);
+
+    /**
+     * 할 일 리스트를 반환합니다.
+     * @return 할 일 리스트
+     */
+    public List<Task> getTaskList() {
+
+        log.info(">>> Access Service 할 일 목록 조회");
+
+        List<Task> taskList = tasksMap.values().stream()
+                .collect(Collectors.toList());
+
+        return taskList;
+
+    }
+
+    /**
+     * 주어진 할 일 ID에 해당하는 할 일 정보를 조회하여 반환합니다.
+     * @param taskId 조회할 할 일 ID
+     * @return 조회한 할 일 정보
+     */
+    public Task getTaskOne(Long taskId) {
+
+        log.info(">>> Access Service 할 일 상세조회");
+
+        Task findTask = tasksMap.get(taskId);
+
+        if( findTask == null ) {
+            throw new TaskNotFoundException(taskId);
+        }
+
+        return findTask;
+
+    }
+
+    /**
+     * 새로운 할 일을 등록합니다.
+     * @param newTask 새로 등록할 할 일
+     * @return 등록한 할 일
+     */
+    public Task saveTask(Task newTask) {
+
+        log.info(">>> Access Service 할 일 등록");
+
+        newTask.setId(taskIdSeq++);
```

Comment:
taskIdSeq의 값을 증가시키고나서 이전의 값을 설정하는 것 같습니다. 명시적으로 값을 수정하는 것과 할당하는 것을 분리하면 좋을 것 같아요.
```suggestion
        newTask.setId(taskIdSeq);
        taskIdSeq += 1;
```

Reply:
값의 증가와 할당을 따로 분리해보니 어떤 로직을 거치는지 더 명확하게 보이는것같습니다.

추기적으로 Id를 구하는 로직을 아래와 같이  별로 메서드로 분리해보았습니다.
![image](https://user-images.githubusercontent.com/78334008/118117069-0de80b00-b426-11eb-9961-41e947f6c043.png)
![image](https://user-images.githubusercontent.com/78334008/118117101-1b04fa00-b426-11eb-8741-7f6b4ef06f2a.png)